dropshot_endpoint/src/error_store.rs

```diff
@@ -0,0 +1,105 @@
+// Copyright 2024 Oxide Computer Company
+
+//! Handle lists of errors in Dropshot.
```

Comment:
Can you explain more of what you're doing and why. Obviously you have my attention with the use of unsafe and RefCell--I'd appreciate some context in order to give it a reasonable review.

Reply:
Give me a bit of time to expand on this, will add a comment and also put in an explanation here. (But at a high level, this is basically an `Rc<RefCell<Vec<Error>>>` without the `Rc`. Doing this requires unsafe to perform lifetime extension, which is sound overall -- and actually a common pattern.)

Reply:
Managed to remove the unsafe (also added `#[forbid(unsafe_code)]`), and added a couple of comments to explain why the `RefCell` is necessary.

Reply:
I'm probably too scared of `unsafe` but thank you for figuring out a way around it. I hope it's not much worse!

Reply:
It's actually better!